src/clr-angular/utils/smart-popover/smart-close-button.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ *
+ */
+
+import { Directive, ElementRef, EventEmitter, Output, HostListener, OnDestroy, AfterViewInit } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { ClrSmartPopoverEventsService } from './providers/smart-popover-events.service';
+import { ClrSmartPopoverToggleService } from './providers/smart-popover-toggle.service';
+import { filter } from 'rxjs/operators';
+
+@Directive({
+  selector: '[clrSmartCloseButton]',
+  host: {
+    '[class.clr-smart-close-button]': 'true',
+  },
+})
+export class ClrSmartCloseButton implements OnDestroy, AfterViewInit {
+  private subscriptions: Subscription[] = [];
+
+  constructor(
+    private elementRef: ElementRef,
+    private smartEventsService: ClrSmartPopoverEventsService,
+    private smartOpenService: ClrSmartPopoverToggleService
+  ) {
+    this.subscriptions.push(
+      smartOpenService.openChange.pipe(filter(value => !value)).subscribe(closeAction => {
+        this.closeChange.next(closeAction);
+      })
+    );
+  }
+
+  @Output('clrSmartCloseChange') closeChange: EventEmitter<boolean> = new EventEmitter<boolean>();
```

Comment:
Do we need this to be Change with boolean param?
My point is that we're only triggering this on close (i.e. with "false"), and whoever subscribes to it will only be interested in the closing event. We won't ever emit other value.
And CloseChange(false) is a bit confusing as definition.
I'd suggest we change it to something like `clrSmart(On)Close` and emit no state (i.e. declare it as void).

Reply:
1) I have usually passed values in the `@Outputs`. I guess I was thinking that if someone wanted to use that value directly it would help them. I do see your point that its only going to fire with flase values. I can change it if you feel strongly about it. 

2) wrt names - I am trying to follow Angular Input/Output naming conventions. `clrSmartCloseChange` seems to make sense to me but I've been looking at this code for a while now. If fresh eyes think its clearer as `clrSmartCoseOnChange` I'm happy to change it. 

let me know your preference for both #1 and #2.

Reply:
I mean, logically `CloseChange(false)` will be interpreted as `not close` :) 
While `Open` can be a state, `Close` is an action. `Closed` might be a state, but it's a bit ambiguous, I think.
As a user, if I want to subscribe to something on a close-handler, it would be the close event itself. In the quality of an action performed. If I'm caring about the state, then I'd be looking for the `Open` state of the popover itself and would subscribe to the state-change events that carry payload about this state. But for an action like `Close`, I only care if it happened.